keeper/src/main/scala/zio/keeper/hyparview/protocols.scala

```diff
@@ -5,35 +5,93 @@ import zio.stm._
 import zio.keeper._
 import zio.stream.ZStream
 import zio.keeper.transport.{ Connection, Protocol }
+import zio.keeper.transport.Transport
 
 object protocols {
 
-  def hyparview[R <: HyParViewConfig with Views with TRandom, E](
-    con: Connection[R, E, Message, Message],
-    peerMessages: Enqueue[(NodeAddress, Message.PeerMessage)]
+  def runInitial[R <: HyParViewConfig with Views with TRandom, E](
+    con: Connection[R, E, Message, Message]
   ): ZIO[R, E, Unit] =
     ZManaged.switchable[R, E, Boolean => UIO[Unit]].use { switch =>
       val protocol =
         initialProtocol
           .contM[R, E, Message, Message, Any] {
             case Some(remoteAddr) =>
               switch(inActiveView(remoteAddr, con.send)).map { setKeepRef =>
-                Right(activeProtocol(remoteAddr, peerMessages.contramap((remoteAddr, _))).onEnd(setKeepRef).unit)
+                Right(activeProtocol(remoteAddr).onEnd(setKeepRef).unit)
               }
             case None =>
               ZIO.succeedNow(Left(()))
           }
-      Protocol.run(con, protocol).unit
+      protocol.run(con).unit
     }
 
+  def runActive[R <: HyParViewConfig with Views with TRandom, E](
+    remoteAddress: NodeAddress,
+    con: Connection[R, E, Message, Message]
+  ): ZIO[R, E, Unit] =
+    inActiveView(remoteAddress, con.send).use { setKeepRef =>
+      activeProtocol(remoteAddress).onEnd(setKeepRef).run(con).unit
+    }
+
+  /**
+   * Start protocol with remote node.
+   * The protocol will be chosen depending on the passed message.
+   * If a neighbor message is sent, assume that this
+   * process is triggered by Views. In this case
+   * start a new doNeighbor process on failure.
+   */
+  def connectRemote(
```

Comment:
It would be nice to avoid introspecting the message here, but I can't think of a nice way that does not result in unnecessary message overhead.